Remotion/Validation/UnitTests/RuleBuilders/AddingObjectValidationRuleBuilderTest.cs

```diff
@@ -14,61 +14,153 @@
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(metaValidationRuleStub.Object);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_FuncOverload ()
     {
+      var fakeValidationResult = MetaValidationRuleValidationResult.CreateValidResult();
+
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(v => fakeValidationResult);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First(), Is.EqualTo(fakeValidationResult));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_ExpressionOverload_IsValidIsTrue ()
     {
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule<IObjectValidator>(v => true);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First().IsValid, Is.True);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_ExpressionOverload_IsValidIsFalse ()
     {
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule<IObjectValidator>(v => false);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First().IsValid, Is.False);
+      Assert.That(rule.Validate(new IObjectValidator[0]).First().Message, Is.EqualTo(
+          "Meta validation rule 'v => False' failed for validator 'Remotion.Validation.Validators.IObjectValidator' on type 'Remotion.Validation.UnitTests.TestDomain.Customer'."));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void SetCondition ()
     {
+      Func<Customer, bool> predicate = _ => true;
+
+      _addingObjectValidationRuleCollectorMock
+          .Setup(mock => mock.SetCondition(predicate))
+          .Verifiable();
+
+      _addingObjectValidationBuilder.SetCondition(predicate);
+
+      _addingObjectValidationRuleCollectorMock
+          .Verify(mock => mock.SetCondition(predicate));
```

Comment:
```suggestion
      _addingObjectValidationRuleCollectorMock.Verify();
```
Unless there's a good reason we need to do it this way, I try to avoid repeating myself. Either, we don't need the setup in line 156 or we do just a generic `Verify` in the assertion block of the test. Repeating the actual call is annoying when you need to update stuff in the future.